frontend/src/Page/Preparation.elm

```diff
@@ -954,8 +1077,24 @@ innerUpdate ctx msg model =
                 ( Preparing form, RemoteData.Success proposalsDict ) ->
                     case Dict.get actionId proposalsDict of
                         Just prop ->
-                            ( { model | pickProposalStep = Done form prop }
-                            , Cmd.none
+                            let
+                                verificationCmd =
+                                    case prop.metadata of
+                                        RemoteData.Success _ ->
+                                            verifyCip100Metadata actionId prop.metadataUrl
+                                                |> Cmd.map ctx.wrapMsg
```

Comment:
Let’s avoid sending a request and make the verify server work for every proposal metada we receive. We can make the request only when a proposal is selected. This will significantly reduce both the traffic and the cpu usage for the verify server.